### PR TITLE
Implementation of breadcrumbs

### PR DESCRIPTION
**Pull request by loichuder**

### CODE REVIEW

File: src/h5web/App.tsx [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
+import BreadcrumbsBar from './BreadcrumbsBar';
 
 function App(): JSX.Element {
-  const [selectedLink, setSelectedLink] = useState<HDF5Link>();
   const [selectedDataset, setSelectedDataset] = useState<HDF5Dataset>();
+  const [selectedNode, setSelectedNode] = useState<TreeNode<HDF5Link>>();
 
   const [isExplorerOpen, setExplorerOpen] = useState(true);
-  const [role, setRole] = useState<Role>(Role.Display);
+  const [isInspecting, setInspecting] = useState(false);
```
Comment (loichuder) [inline src/h5web/App.tsx:19]: Removed `enum` in favour of a boolean

File: src/h5web/providers/models.ts [hunk 2]
```diff
@@ -71,7 +71,7 @@ export enum HDF5LinkClass {
 
 export interface HDF5RootLink {
   class: HDF5LinkClass.Root;
-  title: '';
+  title: string;
```
Comment (loichuder) [inline src/h5web/providers/models.ts:74]: The title of the root link stores now the domain (HDF5 filename)

File: src/h5web/explorer/models.ts [hunk 3]
```diff
@@ -4,4 +4,5 @@ export interface TreeNode<T> {
   level: number;
   data: T;
   children?: TreeNode<T>[];
+  parents: T[];
```
Comment (loichuder) [inline src/h5web/explorer/models.ts:7]: A TreeNode now stores a list of its parents in descending order

File: src/h5web/App.tsx [hunk 4]
```diff
@@ -51,51 +48,21 @@ function App(): JSX.Element {
           flex={isExplorerOpen ? 0.75 : 1}
           minSize={500}
         >
-          <div className={styles.toolbar}>
```
Comment (loichuder) [inline src/h5web/App.tsx:54]: The explorer and role toggler were moved in a dedicated component `BreadcrumbsBar` that also implements the display of the breadcrumbs

File: src/h5web/App.tsx [hunk 5]
```diff
@@ -1,27 +1,24 @@
...
 function App(): JSX.Element {
-  const [selectedLink, setSelectedLink] = useState<HDF5Link>();
   const [selectedDataset, setSelectedDataset] = useState<HDF5Dataset>();
+  const [selectedNode, setSelectedNode] = useState<TreeNode<HDF5Link>>();
 
   const [isExplorerOpen, setExplorerOpen] = useState(true);
-  const [role, setRole] = useState<Role>(Role.Display);
+  const [isInspecting, setInspecting] = useState(false);
 
+  const selectedLink = selectedNode ? selectedNode.data : undefined;
```
Comment (axelboc) [inline src/h5web/App.tsx:21]: `selectedNode && selectedNode.data` ou même `selectedNode?.data`

File: src/h5web/BreadcrumbsBar.module.css [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  padding: calc(0.5rem - 1px) 1rem 0.5rem;
+}
+
+.breadCrumbs {
+  display: flex;
+  align-items: center;
+}
+
+.crumb {
+  margin: 0 0.15rem 0.1rem;
```
Comment (axelboc) [inline src/h5web/BreadcrumbsBar.module.css:15]: I'd typically stick with multiples of 1/8px, i.e. 0.125rem.

File: src/h5web/BreadcrumbsBar.module.css [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+.selected {
```
Comment (axelboc) [inline src/h5web/BreadcrumbsBar.module.css:18]: `data-selected` ?

File: src/h5web/BreadcrumbsBar.module.css [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  composes: crumb;
+  font-weight: 600;
+}
+
+.roleToggler {
```
Comment (axelboc) [inline src/h5web/BreadcrumbsBar.module.css:23]: `modeToggler` ? just a suggestion

File: src/h5web/BreadcrumbsBar.module.css [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  display: flex;
+  border-radius: 0.5rem;
+  overflow: hidden;
+}
+
+.btn {
+  composes: btn-clean from global;
+  display: flex;
+  align-items: center;
+  padding: 0.2rem 0.8rem 0.3rem;
```
Comment (axelboc) [inline src/h5web/BreadcrumbsBar.module.css:33]: Same same, multiples of `0.125rem` if you can.

**Pull request merged**